r/R/util.R

```diff
@@ -110,3 +110,18 @@ handle_embedded_nul_error <- function(e) {
   }
   stop(e)
 }
+
+check_tabular <- function(x, format) {
+  if (!inherits(x, "ArrowTabular")) {
+    abort(
+      c(
+        paste0("Cannot write to ", format, "."),
+        x = paste0(
+          "`x` must be an object of class 'data.frame', 'RecordBatch', or 'Table', not '",
+          class(x)[1],
+          "'"
+        )
+      )
+    )
+  }
+}
```

Comment:
```suggestion
}

```

GitHub seems concerned that this does not have a newline. 

Should this perhaps end with `invisible(x)` ? Probably not important with the current use case, but that would allow e.g. `do_something(check_tabular(x), ...)`

Reply:
Thanks for looking at this- have updated it to do this now to return `invisible(x)` & will update the tests to use `expect_snapshot_error` when making a PR to upgrade to testthat 3e.